### PR TITLE
Issue-12857: Remove ES Dependency from DI Workflow

### LINKED ISSUE #12857: Migrate DI ES sink from ingestion to ES event publisher

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes:

Fixes #12857 
- Introduce `EntityTimeSeriesInterface` and `EntityTimeSeriesRepository` to define Interface for time series record in OM
- Move indexing operation from DI ingestion workflow to backend
- Remove elasticsearch dependencies from ingestion code base
- added data insight sample data ingestion

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [x] I have added tests around the new logic.
- [x] For connector/ingestion changes: I updated the documentation.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/search/SearchEventPublisher.java [hunk 1]
```diff
@@ -68,7 +67,7 @@ public void onShutdown() {
   }
 
   public static void updateElasticSearchFailureStatus(
-      EntityInterface entity, EventPublisherJob.Status status, String failureMessage) {
+      Object entity, EventPublisherJob.Status status, String failureMessage) {
```
Comment (TeddyCr) [inline openmetadata-service/src/main/java/org/openmetadata/service/search/SearchEventPublisher.java:70]: From what I see in that method we are simply doing `entity.toString()` so we should be able to accept any kind of objects

**Review by harshach — APPROVED** [review]

**Comment by harshach** [comment]
@TeddyCr, it looks like we completely removed the ability to run the search indexing externally right? 
@mohityadav766 for the new apps section, SearchIndex scheduled should be internally scheduled app.

**Comment by TeddyCr** [comment]
> @TeddyCr, it looks like we completely removed the ability to run the search indexing externally right?
> @mohityadav766 for the new apps section, SearchIndex scheduled should be internally scheduled app.

That is correct, I believe it was already removed. We just had kept the class for the DI -- @pmbrull to confirm

**Comment by pmbrull** [comment]
> > @TeddyCr, it looks like we completely removed the ability to run the search indexing externally right?
> > @mohityadav766 for the new apps section, SearchIndex scheduled should be internally scheduled app.
> 
> That is correct, I believe it was already removed. We just had kept the class for the DI -- @pmbrull to confirm

The `metadata_elasticsearch` process that was in charge of running the reindex was already triggering the Job on the server, since we had so many misalignments with the indexes definitions

**Pull request merged**